### DISCUSSION TITLE
Extensions and copilot slow to update in offline installation

### DISCUSSION THREAD

**Discussion by christopherpalm264** [coder/coder#11641] [General]
Hi!

I got a very nice setup with coder and everything works fine except an annoying detail. 

It's slow to update extensions and use copilot. It seems to have something to do with some relayed connection that I don't understand. 
I have a coder installation on a kubernetes cluster behind a firewall. I've chosen this configuration below in my values.yaml:
```
    # Only use relayed connections
    - name: "CODER_DERP_SERVER_STUN_ADDRESSES"
      value: "disable"
```
      
Is it possible to get a faster connection without installing your own derp which sounds complicated? 

I appreciate any help!

Thanks,

Chris

**Comment by ammario** [comment]
Generally, Coder doesn't a play a role in facilitating connectivity between the workspace and the internet. Our networking layer comes into play when connecting users to workspaces, e.g. via SSH or `coder port-forward`.

So, if you're seeing slow Copilot and extension updates, there may be some mesh or VPN on your workspace slowing things down. Something to try is manually creating a workspace in your cluster w/o Coder, connecting to it, and seeing if there if there's any change in experience.
